### PR TITLE
bugfix in `sbpy.photometry`

### PR DESCRIPTION
**Pull request by jianyangli**
This PR replaces #286 , which replaced PR #283

Fix different results between model evaluation and direct call to .evaluate method for photometry models

### CODE REVIEW

**Comment by jianyangli** [comment]
@anttipenttila Can you verify that this fix works for you?  I'll merge it once hear back from you.  Thanks.

File: sbpy/photometry/core.py [hunk 1]
```diff
@@ -1026,6 +1026,8 @@ def _G2(self):
 
     @staticmethod
     def evaluate(ph, h, g1, g2):
+        if isinstance(ph, u.Quantity):
+            ph = ph.to('rad').value
```
Comment (mkelley) [inline sbpy/photometry/core.py:1030]: Where is the default behavior (assumed units on `ph`) documented?  Would `ph = u.Quantity(ph, 'deg').to_value('rad')` be acceptable?  That way the default (if it is degrees) is clear in the code and the if statement can be avoided.

Comment (jianyangli) [reply sbpy/photometry/core.py:1030]: Good trick.  Will do it.  Thanks.

**Comment by mkelley** [comment]
Hi @jianyangli @anttipenttila, can this be resolved soon?  Thanks.

**Comment by jianyangli** [comment]
@anttipenttila, I will merge this branch by next Tuesday, July 13.  If I don't hear from you by then, then we can revisit it later if you find other bugs.  Thanks.

**Comment by anttipenttila** [comment]
You can merge, I'm sure you tested with the example I had that showed the bug, and I don't have anything else on this.

**Pull request merged**